pkg/checks/latency/latency.go

```diff
@@ -243,8 +205,6 @@ func (l *Latency) check(ctx context.Context) map[string]result {
 			mu.Lock()
 			defer mu.Unlock()
 
-			l.metrics.duration.DeletePartialMatch(prometheus.Labels{"target": target})
-			l.metrics.duration.WithLabelValues(target, strconv.Itoa(results[target].Code)).Set(results[target].Total)
 			l.metrics.totalDuration.WithLabelValues(target).Set(results[target].Total)
 			l.metrics.count.WithLabelValues(target).Inc()
```

Comment:
Can you take a look here? I think we increment this twice for each target which results in an incorrect count.

Reply:
Indeed:

```shell
# HELP sparrow_latency_count Count of latency checks done
# TYPE sparrow_latency_count counter
sparrow_latency_count{target="https://gitlab.devops.telekom.de"} 7
sparrow_latency_duration_count{target="https://gitlab.devops.telekom.de"} 4
```

And now with the change, the two metrics (which should always have the same value) go hand-in-hand:

```shell
# HELP sparrow_latency_count Count of latency checks done
# TYPE sparrow_latency_count counter
sparrow_latency_count{target="https://gitlab.devops.telekom.de"} 9
sparrow_latency_duration_count{target="https://gitlab.devops.telekom.de"} 9
```

All checks now have the same count, as expected (when the interval of each check is the same):


```shell
# HELP sparrow_dns_check_count Total number of DNS checks performed on the target and if they were successful.
# TYPE sparrow_dns_check_count counter
sparrow_dns_check_count{target="gitlab.devops.telekom.de"} 13
sparrow_dns_duration_count{target="gitlab.devops.telekom.de"} 13
# HELP sparrow_latency_count Count of latency checks done
# TYPE sparrow_latency_count counter
sparrow_latency_count{target="https://gitlab.devops.telekom.de"} 13
sparrow_latency_duration_count{target="https://gitlab.devops.telekom.de"} 13
```